porcupine/plugins/filemanager.py

```diff
@@ -122,6 +157,31 @@ def rename(old_path: Path) -> None:
         tab.path = new_path / tab.path.relative_to(old_path)
 
 
+def paste(new_path: Path) -> None:
+    assert copy_path is not None
+    new_file_path = new_path / copy_path.name
+
+    if new_file_path.exists():
+        if new_file_path.parent == copy_path.parent:  # copying file in the same dir (sequential number copy)
+            path_number = 0
+            path_stem = new_file_path.stem
+            path_suffix = new_file_path.suffix
+            while new_file_path.exists():
+                path_number += 1
+                new_file_path = new_file_path.with_name(path_stem + f" ({path_number})" + path_suffix)
+        else:
+            new_file_path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
+
+    if new_file_path is not None:
+        new_file_path.touch()
+        shutil.copy(copy_path, new_file_path)
```

Comment:
The new file does not appear in the directory tree when the copying is done. I think adding `get_directory_tree().refresh()` should fix it.

Reply:
This is strange, in my case it automatically refreshes without the instruction

Reply:
It depends on how focusing events work with the window manager you use. It refreshes when the Porcupine window is focused:

https://github.com/Akuli/porcupine/blob/719da27426350b503c910850fb757716b366efe0/porcupine/plugins/directory_tree.py#L442